### DISCUSSION TITLE
Can I create a CSS string using `editor.Css.getRules()`

### DISCUSSION THREAD

**Discussion by FaisalShaikhHA** [GrapesJS/grapesjs#4749] [Q&A] [answered]
Can I create a CSS string using `editor.Css.getRules()` 
```
for (let rule of editor.Css.getRules()) {
      if (rule.getComponent()) {
          // concat below string and save to db
          rule.toCSS()
      }
}
```
will the above CSS string be equivalent to CSS used on components only or editor.getCss()?

**Comment by artf** [comment, marked as answer]
That will return all component related CSS styles (those created via Style Manager when `selectorManager.componentFirst` is enabled)

**Reply by FaisalShaikhHA** [reply]
Hi @artf, how should I get all the CSS related to components and skip unwanted CSS which does not belong to any component?
Because I am getting unwanted CSS when I call the `editor.getCss()` here is that discussion [link](https://github.com/artf/grapesjs/discussions/4747#discussion-4601850) which describes my issue.

**Reply by artf** [reply]
You can use [`getCss`](https://grapesjs.com/docs/api/editor.html#getcss) with `component` and `onlyMatched` options

**Reply by FaisalShaikhHA** [reply]
Thank you @artf for the help, I have this `*, *::before, *::after{box-sizing:inherit;}` style defined on my page. 
When I use the below code I get this `*, *::before, *::after{box-sizing:inherit;}` along with component style. How can I avoid this style?
 ```
editor.getCss({
      component: editableComponent,
      onlyMatched: true,
  });
```

**Reply by artf** [reply]
well that style always matches any element so you if you need you can remove it manually from the output.
